### PR TITLE
6213 Сахарова Дарья Lab5 var_7

### PR DESCRIPTION
**Pull request by daryaskhrv**
1. Разделен набор данных на сбалансированные выборки,
2. При помощи PyTorch реализована модель нейронной сети,
3. Обучена модель и построены графики.

ссылка на google collaboratory:
https://colab.research.google.com/drive/1eswJKpzGUWl-DeJ0V1L0TBc_8rDaRySY?usp=sharing

### CODE REVIEW

File: Lab5/main.ipynb [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+    "4:  Пайплайн предобработки данных "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 67,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_transforms =  transforms.Compose([\n",
```
Comment (alxmcs) [inline Lab5/main.ipynb:189]: Преобразования всех трех выборок аналогичны, заводить 3 преобразования на них излишне

File: Lab5/main.ipynb [hunk 2]
```diff
@@ -0,0 +1,709 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images_list = []\n",
+    "images_list = glob.glob(os.path.join('C:/Users/user/Desktop/dataset1','*.jpg'))\n",
```
Comment (alxmcs) [inline Lab5/main.ipynb:55]: А почему изображений 292, а не 1000?

File: Lab5/main.ipynb [hunk 3]
```diff
@@ -0,0 +1,709 @@
...
+    "6:  Построение графиков"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 76,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def show_results(acc, loss):\n",
```
Comment (alxmcs) [inline Lab5/main.ipynb:573]: Чтобы иметь возможность визуально оценить качество обучения модели, кривые для функции потерь на обучающей и валидационной выборки должны быть изображены на одном графике.
Кривые точности классификации аналогично .

File: Lab5/main.ipynb [hunk 4]
```diff
@@ -0,0 +1,709 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images_list = []\n",
+    "images_list = glob.glob(os.path.join('C:/Users/user/Desktop/dataset1','*.jpg'))\n",
+    "train_list = images_list[0 : int(len(images_list)*0.8)]\n",
```
Comment (alxmcs) [inline Lab5/main.ipynb:56]: Не вижу проверки на то, что датасеты сбалансированы.
Судя по результатам обучения, они и не сбалансированы - в обучающую выборку отгрузились все лошади и часть зебр, а на тестовую и валидационную выборку остались только зебры.

**Review by alxmcs — CHANGES_REQUESTED** [review]

**Comment by daryaskhrv** [comment]
Новый колаб: https://colab.research.google.com/drive/1MtXSDeB5lWQUub3KGpBQ1kuDYpzcwdVv?usp=sharing

**Review by alxmcs — CHANGES_REQUESTED** [review]
С учетом того, что в модели стоит мощный дропаут `nn.Dropout(0.5)`, то, что точность на вариационной выборке оказалась выше, чем на обучающей, в целом не сильно удивительно.

Но мне кажется, судя по кривым обучения, что модель осталась недоученной. Я бы предложил увеличить число эпох и посмотреть, что из этого выйдет.

**Comment by daryaskhrv** [comment]
https://colab.research.google.com/drive/1KMAJadIluY-6NEjOznHZHTkSsxSEDP6E?usp=sharing

**Review by alxmcs — APPROVED** [review]
Судя по кривым, где-то после 16 итерации модель начинает переобучаться.
Но, думаю,  в таком виде можно принять работу.
